docs/search_paths.md

```diff
@@ -0,0 +1,252 @@
+with `find_package(MyProject)` as long as it is in the `build-system.requires`
+list.
+
+You can use `search.roots` to override or extend these variables, e.g. you can
+delete the `MyProject_ROOT` variable by setting
+
+````{tab} pyproject.toml
+
+```toml
+[tool.scikit-build.search.roots]
+MyProject = ""
+```
+
+````
+
+`````{tab} config-settings
+
+
+````{tab} pip
+
+```console
+$ pip install . -v --config-settings=search.roots.MyProject=""
+```
+
+````
+
+````{tab} build
+
+```console
+$ pipx run build --wheel -Csearch.roots.MyProject=""
+```
+
+````
+
+````{tab} cibuildwheel
+
+```toml
+[tool.cibuildwheel.config-settings]
+"search.roots.MyProject" = ""
+```
+
+````
+
+`````
+
+````{tab} Environment
+
+
+```yaml
+SKBUILD_SEARCH_ROOTS_MyProject: ""
+```
+
+````
+
+Or you could point to another preferred path:
+
+
+````{tab} pyproject.toml
+
+```toml
+[tool.scikit-build.search.roots]
+MyProject = "other_project"
+```
+
+````
+
+`````{tab} config-settings
+
+
+````{tab} pip
+
+```console
+$ pip install . -v --config-settings=search.roots.MyProject="other_project"
+```
+
+````
+
+````{tab} build
+
+```console
+$ pipx run build --wheel -Csearch.roots.MyProject="other_project"
+```
+
+````
+
+````{tab} cibuildwheel
+
+```toml
+[tool.cibuildwheel.config-settings]
+"search.roots.MyProject" = "other_project"
+```
+
+````
+
+`````
+
+````{tab} Environment
+
+
+```yaml
+SKBUILD_SEARCH_ROOTS_MyProject: "other_project"
+```
+
+````
+
+## `CMAKE_PREFIX_PATH`
+
+Another common search path that scikit-build-core populates is the
+`CMAKE_PREFIX_PATH` which is a common catch-all for all CMake search paths,
+e.g. `find_package`, `find_program`, `find_path`. This is populated by default
+with the install and build path's `site-packages` folder, but this can be
+disabled by setting
+
+```toml
+[tool.scikit-build.search]
+search.use-install-prefix = false
+search.use-build-prefix = false
+```
+
+Additionally, scikit-build-core reads the entry-point `cmake.prefix` and the
+option `search.prefixes` to prepend to this path list. You can similarly export
+this as
+
+``` toml
+[project.entry-points."cmake.prefix"]
+MyProject = "myproject"
+```
+
+and you can similarly delete this entry-point variable
+
+````{tab} pyproject.toml
+
+```toml
+[tool.scikit-build.search.prefixes]
+MyProject = ""
+```
+
+````
+
+`````{tab} config-settings
+
+
+````{tab} pip
+
+```console
+$ pip install . -v --config-settings=search.prefixes.MyProject=""
+```
+
+````
+
+````{tab} build
+
+```console
+$ pipx run build --wheel -Csearch.prefixes.MyProject=""
+```
+
+````
+
+````{tab} cibuildwheel
+
+```toml
+[tool.cibuildwheel.config-settings]
+"search.prefixes.MyProject" = ""
+```
+
+````
+
+`````
+
+````{tab} Environment
+
+
+```yaml
+SKBUILD_SEARCH_PREFIXES_MyProject: ""
+```
+
+````
+
+If you only wish to expand the prefix path, you can also define
+`search.prefixes` as a list
+
+```toml
+[tool.scikit-build.search]
+prefixes = ["/path/to/prefixA", "/path/to/prefixB"]
```

Comment:
Yes, these are just paths. You can replace this with the following one line in CMakeLists.txt:

```cmake
list(APPEND CMAKE_PREFIX_PATH /path/to/prefixA /path/to/prefixB)
```

We should only add features hard to implement in CMake.

Reply:
I think it would be good to support this anyway because it could be used via overrides and cli. Similarly for the entry-point having a `path` as value instead of an importable python reference

Reply:
Entry points are supposed to be modules or objects in modules, not arbitrary strings. The syntax is `<module>[:<attribute>]`.

Reply:
And you can still do it via overrides and CLI using defines.

Reply:
Hmm, I guess deleting the entry-point would also work. I was considering when packaging for Fedora and we would want the entry-point to point to the system path in `/usr/local`. But I guess there are other issues with that since the entry-points are defined when building the sdist.

What about checking if the entry-point points to a `str` or `pathlib.Path`/`traversable` or `list` of those. If not use the current logic of loading `importlib.resources.files`

Reply:
Hmm, this point needs to be revisited. If the user sets `cmake.define.CMAKE_PREFIX_PATH`, who would win? Other than that, shall we keep the list definition like this? It can be useful for setting via cli and resolving the ambiguity with scikit-build-core also setting that variable.